proto/bandchain/v1/oracle/proof.proto

```diff
@@ -0,0 +1,173 @@
+
+import "google/api/annotations.proto";
+import "gogoproto/gogo.proto";
+import "oracle/v1/oracle.proto";
+
+// Service defines the gRPC querier service.
+service Service {
+  // Proof queries the proof for given request ID
+  rpc Proof(QueryProofRequest) returns (QueryProofResponse) {
+    option (google.api.http).get = "/bandchain/v1/oracle/proof/{request_id}";
+  }
+
+  // MultiProof queries multiple proofs for given list of request IDs
+  rpc MultiProof(QueryMultiProofRequest) returns (QueryMultiProofResponse) {
+    option (google.api.http).get = "/bandchain/v1/oracle/multi_proof";
+  }
+
+  // RequestCountProof queries the count proof
+  rpc RequestCountProof(QueryRequestCountProofRequest) returns (QueryRequestCountProofResponse) {
+    option (google.api.http).get = "/bandchain/v1/oracle/requests_count_proof";
+  }
+}
+
+// QueryProofRequest is request type for the Service/Proof RPC method.
+message QueryProofRequest {
+    // RequestID is ID of an oracle request
+    uint64 request_id = 1;
+    // height is block height
+    int64 height = 2;
+}
+
+// QueryChainIDResponse is response type for the Service/Proof RPC method.
+message QueryProofResponse {
+  // height is block height
+  int64 height = 1;
+  // result is the proof
+  SingleProofResponse result = 2;
+}
+
+// QueryMultiProofRequest is request type for the Service/MultiProof RPC method.
+message QueryMultiProofRequest {
+  // request_ids is the list of request IDs
+  repeated uint64 request_ids = 1;
+}
+
+// QueryMultiProofResponse is response type for the Service/MultiProof RPC method.
+message QueryMultiProofResponse {
+  int64 height = 1;
+  MultiProofResponse result = 2;
+}
+
+// QueryRequestCountProofRequest is request type for the Service/RequestCountProof RPC method.
+message QueryRequestCountProofRequest {}
+
+// QueryRequestCountProofResponse is response type for the Service/RequestCountProof RPC method.
+message QueryRequestCountProofResponse {
+  int64 height = 1;
+  CountProofResponse result = 2;
+}
+
+// SingleProofResponse is the data structure for response of single proof
+message SingleProofResponse {
+  SingleProof proof = 1;
+  bytes evm_proof_bytes = 2;
+}
+
+// MultiProofResponse is the data structure for response of multi proof
+message MultiProofResponse {
+  MultiProof proof = 1;
+  bytes evm_proof_bytes = 2;
+}
+
+// CountProofResponse is the data structure for response of count proof
+message CountProofResponse {
+  CountProof proof = 1;
+  bytes evm_proof_bytes = 2;
+}
+
+// SingleProof contains block height, oracle data proof and block relay proof
+message SingleProof {
+  uint64 block_height = 1;
+  OracleDataProof oracle_data_proof = 2;
+  BlockRelayProof block_relay_proof = 3;
+}
+
+// MultiProof contains block height, list of oracle data proof and block relay proof
+message MultiProof {
+  uint64 block_height = 1;
+  repeated OracleDataProof oracle_data_multi_proof = 2;
```

Comment:
Please add `[ (gogoproto.nullable) = false ]` to generate `[]OracleDataProof` not `[]*OracleDataProof`